### PR TITLE
sqlite3: add 3.35.1 + fix exec install on iOS + provide official CMake variables

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **sqlite3/3.35.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jgsogo — APPROVED** [review]

**Comment by paulo-coutinho** [comment]
Hi,

Please, add an option to enable new resource added in version 3.35.0:
https://sqlite.org/releaselog/3_35_0.html

Option:
```enable_math_functions```
Default:
```False```

Args added to compile:
```-DSQLITE_ENABLE_MATH_FUNCTIONS```

Thanks.

**Comment by SpaceIm** [comment]
Thanks @paulo-coutinho, I missed that. Looking at documentation, shouldn't this option be True by default?

**Comment by paulo-coutinho** [comment]
In my opinion no, because it is optional by default. It is disabled because is a new feature.

See:
https://sqlite.org/lang_mathfunc.html

```
The math functions shown below are part of the SQLite amalgamation source file but are only active if the amalgamation is compiled using the -DSQLITE_ENABLE_MATH_FUNCTIONS compile-time option.
```

It is not active by default.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — DISMISSED** [review]

**Review by a4z — DISMISSED** [review]

**Comment by SpaceIm** [comment]
> 
> 
> In my opinion no, because it is optional by default. It is disabled because is a new feature.
> 
> See:
> https://sqlite.org/lang_mathfunc.html
> 
> ```
> The math functions shown below are part of the SQLite amalgamation source file but are only active if the amalgamation is compiled using the -DSQLITE_ENABLE_MATH_FUNCTIONS compile-time option.
> ```
> 
> It is not active by default.

Actually it's enabled by default according to documentation:
https://sqlite.org/compile.html#enable_math_functions

And in configure if you download https://www.sqlite.org/2021/sqlite-autoconf-3350000.tar.gz:

```shell
# Check whether --enable-math was given.
if test "${enable_math+set}" = set; then :
  enableval=$enable_math;
else
  enable_math=yes
fi

{ $as_echo "$as_me:${as_lineno-$LINENO}: checking SQL math functions" >&5
$as_echo_n "checking SQL math functions... " >&6; }
if test x"$enable_math" = "xyes"; then
  BUILD_CFLAGS="$BUILD_CFLAGS -DSQLITE_ENABLE_MATH_FUNCTIONS"
  { $as_echo "$as_me:${as_lineno-$LINENO}: result: enabled" >&5
$as_echo "enabled" >&6; }
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing ceil" >&5
$as_echo_n "checking for library containing ceil... " >&6; }
if ${ac_cv_search_ceil+:} false; then :
  $as_echo_n "(cached) " >&6
else
  ac_func_search_save_LIBS=$LIBS
cat confdefs.h - <<_ACEOF >conftest.$ac_ext
```

**Comment by SpaceIm** [comment]
See https://cmake.org/cmake/help/latest/module/FindSQLite3.html for official variables.

File: recipes/sqlite3/all/CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,9 @@ option(ENABLE_UNLOCK_NOTIFY "Enable support for the unlock notify API")
 option(ENABLE_DEFAULT_SECURE_DELETE "Turns on secure deletion by default")
 option(USE_ALLOCA "The alloca() memory allocator will be used in a few situations where it is appropriate.")
 option(OMIT_LOAD_EXTENSION "Omits the entire extension loading mechanism from SQLite")
+if(SQLITE3_VERSION VERSION_GREATER_EQUAL "3.35.0")
```
Comment (SSE4) [inline recipes/sqlite3/all/CMakeLists.txt:24]: VERSION_GREATER_EQUAL  requires CMake 3.7, may you add check?

File: recipes/sqlite3/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -5,6 +5,10 @@ include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
 find_package(SQLite3 REQUIRED)
+message(STATUS "SQLite3_INCLUDE_DIRS: ${SQLite3_INCLUDE_DIRS}")
```
Comment (SSE4) [inline recipes/sqlite3/all/test_package/CMakeLists.txt:8]: is it necessary? might be noizy

Comment (SpaceIm) [reply recipes/sqlite3/all/test_package/CMakeLists.txt:8]: Not really, it was a test, I can remove those lines.

File: recipes/sqlite3/all/CMakeLists.txt (hunk 1 above)
Comment (SpaceIm) [reply recipes/sqlite3/all/CMakeLists.txt:24]: good catch, let's bump min version to 3.7 (4 years old version, sounds reasonable)

**Review by SSE4 — APPROVED** [review]

**Comment by SpaceIm** [comment]
CI is sick.

File: recipes/sqlite3/all/conanfile.py [hunk 3]
```diff
@@ -131,13 +170,15 @@ def package_info(self):
         self.cpp_info.names["cmake_find_package_multi"] = "SQLite"
         self.cpp_info.components["sqlite"].names["cmake_find_package"] = "SQLite3"
         self.cpp_info.components["sqlite"].names["cmake_find_package_multi"] = "SQLite3"
+        self.cpp_info.components["sqlite"].builddirs.append(self._module_subfolder)
+        self.cpp_info.components["sqlite"].build_modules["cmake_find_package"] = [self._module_file_rel_path]
```
Comment (prince-chrismc) [inline recipes/sqlite3/all/conanfile.py:174]: No multi support?

Comment (SpaceIm) [reply recipes/sqlite3/all/conanfile.py:174]: I don't think that `cmake_find_package_multi` defines the variables which are aliased.
Anyway, you should never rely on `cmake_find_package_multi` for libraries with built-in Find module files (like sqlite3, zlib, libcurl, openssl, bzip2, libjpeg), if you call find_package without CONFIG keyword.

**Comment by paulo-coutinho** [comment]
There is bugfixes for version 3.35.0 that is version 3.35.1:

https://sqlite.org/releaselog/3_35_1.html

Can you add it?

**Review by prince-chrismc — DISMISSED** [review]

**Review by a4z — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by mathbunnyru — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

**Review by a4z — APPROVED** [review]

**Review by mathbunnyru — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**